components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -240,27 +240,30 @@ class Session(
      * Set when a load request is received, indicating if the request came from web content, or via a redirect.
      */
     var loadRequestMetadata: Consumable<LoadRequestMetadata>
-        by Delegates.vetoable(Consumable.empty()) { _, _, request ->
-            if (request.peek() == null) {
-                false
-            } else {
-                val new = request.peek() as LoadRequestMetadata
-                val consumers = wrapConsumers<LoadRequestMetadata> {
-                    onLoadRequest(
-                        this@Session,
-                        new.url,
-                        new.isSet(LoadRequestOption.REDIRECT),
-                        new.isSet(LoadRequestOption.WEB_CONTENT)
-                    )
-                }
-
-                val consumed = request.consumeBy(consumers)
-                when (consumed) {
-                    true -> new.allowOrDeny(false)
-                    false -> new.allowOrDeny(true)
+        by Delegates.vetoable(Consumable.empty()) { _, oldRequest, newRequest ->
+            val new = newRequest.peek()
+
+            when {
+                oldRequest == newRequest -> false
```

Comment:
We talked about my concern here that this line essentially makes the crash go away, but we won't know what caused it. Let's sit down and see how we can catch the error and log more information?

Reply:
Agreed.  Thanks!